Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -796,9 +801,115 @@ setActivity (personId, merchantId, merchantOpCityId) isActive mode = do
               QDriverInformation.updateBlockedState driverId False (Just "AUTOMATICALLY_UNBLOCKED") merchantId merchantOpCityId DTDBT.Application
             else throwError $ DriverAccountBlocked (BlockErrorPayload driverInfo.blockExpiryTime driverInfo.blockReasonFlag)
         Nothing -> throwError $ DriverAccountBlocked (BlockErrorPayload driverInfo.blockExpiryTime driverInfo.blockReasonFlag)
-  when (driverInfo.active /= isActive || driverInfo.mode /= mode) $ QDriverInformation.updateActivity isActive (mode <|> Just DriverInfo.OFFLINE) driverId
+  when (driverInfo.active /= isActive || driverInfo.mode /= mode) $ do
+    QDriverInformation.updateActivity isActive (mode <|> Just DriverInfo.OFFLINE) driverId
+    fork "update driver online duration" . Redis.whenWithLockRedis (updateDriverOnlineDurationLockKey driverId) 60 $
+      processingChangeOnline (driverId, merchantId, merchantOpCityId) transporterConfig.timeDiffFromUtc driverInfo mode
   pure APISuccess.Success
 
+updateDriverOnlineDurationLockKey :: Id SP.Person -> Text
+updateDriverOnlineDurationLockKey id = "DriveOnlineDuration:PersonId-" <> id.getId
+
+processingChangeOnline ::
+  (CacheFlow m r, EsqDBFlow m r) =>
+  (Id SP.Person, Id DM.Merchant, Id DMOC.MerchantOperatingCity) ->
+  Seconds ->
+  DriverInfo.DriverInformation ->
+  Maybe DriverInfo.DriverMode ->
+  m ()
+processingChangeOnline (driverId, merchantId, merchantOpCityId) timeDiffFromUtc driverInfo mode = do
+  localTime <- getLocalCurrentTime timeDiffFromUtc
+  let previousMode = driverInfo.mode
+      merchantLocalDate = utctDay localTime
+  now <- getCurrentTime
+  mbDailyStats <- SQDS.findByDriverIdAndDate driverId merchantLocalDate
+  when (previousMode == Just DriverInfo.ONLINE) $ do
+    let mbLastOnlineFrom = addUTCTime (secondsToNominalDiffTime timeDiffFromUtc) <$> driverInfo.onlineDurationRefreshedAt
+        numDaysAgo = 10
```

Comment:
Could you add logs if this limit was reached. it will help debug if any edge cases will be there